# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

In [113]:

SELECT TOP 5 *
FROM annual_2016;

SELECT TOP 5 *
FROM january_2017;

SELECT TOP 5 *
FROM industry;

SELECT TOP 5 *
FROM seasonal;

SELECT TOP 5 *
FROM series;

SELECT *
FROM datatype;


SELECT TOP 5 *
FROM footnote;

SELECT *
FROM period;

SELECT *
FROM supersector;


(5 rows affected)

(5 rows affected)

(5 rows affected)

(2 rows affected)

(5 rows affected)

(45 rows affected)

(2 rows affected)

(13 rows affected)

(22 rows affected)

Total execution time: 00:00:00.059

id series_id year period value footnote_codes original_file 0 CEU5500000007 2016 M13 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CEU5500000008 2016 M13 26.11 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CEU5500000030 2016 M13 962.73 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CEU5500000031 2016 M13 411.29 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CEU5500000032 2016 M13 11.15 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv

id series_id year period value footnote_codes original_file 0 CES5500000007 2017 M01 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CES5500000008 2017 M01 26.27 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CES5500000030 2017 M01 969.36 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CES5500000031 2017 M01 406.67 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CES5500000032 2017 M01 11.02 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv

id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence 0 0 - B Total nonfarm 0 T 1 1 5000000 - A Total private 1 T 2 2 6000000 - A Goods-producing 1 T 3 3 7000000 - B Service-providing 1 T 4 4 8000000 - A Private service-providing 1 T 5

industry_code seasonal_text S Seasonally Adjusted U Not Seasonally Adjusted

series_id supersector_code industry_code data_type_code seasonal series_title CES0000000001 0 00000000 1 S All employees CES0000000010 0 00000000 10 S Women employees CES0000000025 0 00000000 25 S All employees CES0000000026 0 00000000 26 S All employees CES0500000001 5 05000000 1 S All employees

data_type_code data_type_text 1 ALL EMPLOYEES 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES 4 AVERAGE WEEKLY OVERTIME HOURS OF ALL EMPLOYEES 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 9 AVERAGE WEEKLY OVERTIME HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 10 WOMEN EMPLOYEES 11 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES 12 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES 13 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES 15 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES 16 INDEXES OF AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES 17 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES 19 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES 20 AVERAGE WEEKLY OVERTIME HOURS OF ALL EMPLOYEES 21 DIFFUSION INDEXES 22 DIFFUSION INDEXES 23 DIFFUSION INDEXES 24 DIFFUSION INDEXES 25 ALL EMPLOYEES 26 ALL EMPLOYEES 30 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 31 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 32 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 33 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 34 INDEXES OF AGGREGATE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 35 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 36 AVERAGE WEEKLY HOURS 37 AVERAGE WEEKLY OVERTIME HOURS 38 PRODUCTION AND NONSUPERVISORY EMPLOYEES-TO-ALL EMPLOYEES RATIO 39 WOMEN EMPLOYEES-TO-ALL EMPLOYEES RATIO 56 AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES 57 AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES 58 AGGREGATE WEEKLY OVERTIME HOURS OF ALL EMPLOYEES 81 AGGREGATE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 82 AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 83 AGGREGATE WEEKLY OVERTIME HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 86 1-MONTH CONFIDENCE INTERVAL 87 3-MONTH CONFIDENCE INTERVAL 88 6-MONTH CONFIDENCE INTERVAL 89 12-MONTH CONFIDENCE INTERVAL 98 CPI-U 1982-84 99 CPI-W 1982-84

footnote_code footnote_text I Seasonally Adjusted Independently. See https://www.bls.gov/web/empsit/cestn.htm#SA_ind for details. P preliminary

period_code month_abbr month M01 JAN January M02 FEB February M03 MAR March M04 APR April M05 MAY May M06 JUN June M07 JUL July M08 AUG August M09 SEP September M10 OCT October M11 NOV November M12 DEC December M13 AN AV Annual Average

supersector_code supersector_name 0 Total nonfarm 5 Total private 6 Goods-producing 7 Service-providing 8 Private service-providing 10 Mining and logging 20 Construction 30 Manufacturing 31 Durable Goods 32 Nondurable Goods 40 Trade 41 Wholesale trade 42 Retail trade 43 Transportation and warehousing 44 Utilities 50 Information 55 Financial activities 60 Professional and business services 65 Education and health services 70 Leisure and hospitality 80 Other services 90 Government

```
/Users/nicolette/Desktop/LaunchCode/ClassWork/Assignment3_9to5-main/graphic.jpg

```

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees? <b>10</b></span>

In [6]:
SELECT TOP 5 series_title, data_type_code
FROM series
WHERE series_title = 'Women employees'
GROUP BY series_title, data_type_code

(1 row affected)

Total execution time: 00:00:00.053

series_title data_type_code Women employees 10

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector? </span> <span style="font-size: 13.3333px; white-space: pre-wrap;"><b>CES5552211010</b></span>

In [20]:
--Find the supersector_code for 'financial activities'
SELECT supersector_name, supersector_code
FROM supersector
WHERE 
supersector_name = 'financial activities';

--Find the industry code for 'commerical banking'
SELECT industry_name, industry_code
FROM industry
WHERE 
industry_name = 'commercial banking'

--Use the supersector_code, industry_code and data_type_code to find the series_id for women employees in the commercial banking industry in the financial activities supersector.
SELECT TOP 5 series_id, supersector_code, industry_code, data_type_code
FROM series
WHERE supersector_code = 55 AND industry_code = 55522110 AND data_type_code = 10


(1 row affected)

(1 row affected)

(2 rows affected)

Total execution time: 00:00:00.051

supersector_name supersector_code Financial activities 55

industry_name industry_code Commercial banking 55522110

series_id supersector_code industry_code data_type_code CES5552211010 55 55522110 10 CEU5552211010 55 55522110 10

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [12]:
SELECT TOP 10 a.id, a.series_id, a.year, a.period, a.value, a.footnote_codes, a.original_file
FROM annual_2016 AS a
LEFT JOIN series AS s
ON a.series_id = s.series_id;


(10 rows affected)

Total execution time: 00:00:00.161

id series_id year period value footnote_codes original_file 0 CEU5500000007 2016 M13 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CEU5500000008 2016 M13 26.11 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CEU5500000030 2016 M13 962.73 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CEU5500000031 2016 M13 411.29 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CEU5500000032 2016 M13 11.15 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 5 CEU5500000034 2016 M13 111.6 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 6 CEU5500000035 2016 M13 179.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 7 CEU5500000081 2016 M13 236997 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 8 CEU5500000082 2016 M13 6189003 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 9 CEU5552200007 2016 M13 37.3 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [28]:
SELECT TOP 5 *
FROM series AS s
LEFT JOIN datatype AS dt
ON s.data_type_code = dt.data_type_code


(5 rows affected)

Total execution time: 00:00:00.032

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [13]:
SELECT TOP 5 *
FROM series AS s
JOIN industry AS i
ON s.industry_code = i.industry_code

(5 rows affected)

Total execution time: 00:00:00.051

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please. <b>2,340,612</b></span>

In [27]:
SELECT SUM(value) AS 'number_employees'
FROM annual_2016 AS a
LEFT JOIN series AS s
ON a.series_id = s.series_id
WHERE series_title = 'ALL EMPLOYEES';

(1 row affected)

Total execution time: 00:00:00.054

number_employees 2340612.2000000016

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;<b>1,125,490</b></span>

In [15]:
SELECT SUM(value) AS 'number_Women_employees'
FROM annual_2016 AS a
LEFT JOIN series AS s
ON a.series_id = s.series_id
WHERE data_type_code = 10;

(1 row affected)

Total execution time: 00:00:00.057

number_Women_employees 1125489.6000000008

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;<b>1,263,650</b></span>

In [48]:
SELECT SUM(value) AS 'number_employees'
FROM annual_2016 AS a
JOIN series AS s
ON a.series_id = s.series_id
WHERE series_title = 'PRODUCTION AND NONSUPERVISORY EMPLOYEES'

(1 row affected)

Total execution time: 00:00:00.068

number_employees 1263649.6000000008

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?<b> 36.06 hours a week</b></span>

In [56]:
SELECT AVG (value) AS 'avg_weekly_hours_production_nonsupervisory_employees'
FROM january_2017 AS j
JOIN series AS s
ON j.series_id = s.series_id
WHERE series_title = 'Average weekly hours of production and nonsupervisory employees';

SELECT AVG (value) AS 'avg_weekly_hours_all employees'
FROM january_2017 AS j
JOIN series AS s
ON j.series_id = s.series_id
WHERE series_title = 'Average weekly hours of all employees';

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.092

avg_weekly_hours_production_nonsupervisory_employees 36.05862068965521

avg_weekly_hours_all employees 36.38327974276531

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please. </span> <span style="font-size: 13.3333px; white-space: pre-wrap;"><b>1,838,753,220</b></span>

In [106]:
SELECT *
FROM datatype
WHERE data_type_text = 'Aggregate weekly payrolls of production and nonsupervisory employees';

SELECT SUM (value) AS'total weekly payroll'
FROM January_2017 AS j 
JOIN series AS s
ON j.series_id = s.series_id
WHERE series_title = 'Aggregate weekly payrolls of production and nonsupervisory employees';

SELECT SUM (value) AS'total weekly payroll_January_2017'
FROM January_2017 AS j 
JOIN series AS s
ON j.series_id = s.series_id
JOIN datatype AS d
ON d.data_type_code = s.data_type_code
WHERE d.data_type_code = 82;

(1 row affected)

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.122

data_type_code data_type_text 82 AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES

total weekly payroll 1838753220

total weekly payroll_January_2017 1838753220

6. <span style="background-color: rgb(255, 255, 255);"><font color="#666666" face="Arvo, serif"><span style="font-size: 10pt; white-space: pre-wrap;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? </span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>Motor vehicle power train components. </b></span><span style="font-size: 10pt; white-space: pre-wrap;">Which industry was the lowest? </span></font></span> **<span style="font-size: 13.3333px; white-space: pre-wrap;">Fitness and recreational sports centers</span>**

In [107]:
SELECT TOP 4 i.industry_name, i.industry_code, j.value, s.series_id
FROM industry AS i
JOIN series AS s
ON i.industry_code = s.industry_code
JOIN january_2017 AS j 
ON s.series_id = j.series_id
WHERE series_title = 'Average weekly hours of production and nonsupervisory employees' 
ORDER BY j.value DESC;

SELECT TOP 4 i.industry_name, i.industry_code, j.value, s.series_id
FROM industry AS i
JOIN series AS s
ON i.industry_code = s.industry_code
JOIN january_2017 AS j 
ON s.series_id = j.series_id
WHERE series_title = 'Average weekly hours of production and nonsupervisory employees' 
ORDER BY j.value ASC;

(4 rows affected)

(4 rows affected)

Total execution time: 00:00:00.104

industry_name industry_code value series_id Motor vehicle power train components 31336350 49.8 CES3133635007 Motor vehicle power train components 31336350 49.8 CES3133635007 Motor vehicle power train components 31336350 49.4 CEU3133635007 Motor vehicle power train components 31336350 49.4 CEU3133635007

industry_name industry_code value series_id Fitness and recreational sports centers 70713940 16.7 CEU7071394007 Fitness and recreational sports centers 70713940 16.7 CEU7071394007 Fitness and recreational sports centers 70713940 17 CES7071394007 Fitness and recreational sports centers 70713940 17 CES7071394007

7. <span style="background-color: rgb(255, 255, 255);"><font color="#666666" face="Arvo, serif"><span style="font-size: 10pt; white-space: pre-wrap;">In January 2017, which industry was the total weekly payroll for production and nonsupervisory employees the highest? <b>Total Private at $</b></span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>74,498,171. </b></span><span style="font-size: 10pt; white-space: pre-wrap;">Which industry was the lowest? <b>Coin-operated laundries and dry cleaners at $10,079</b></span></font></span>

In [133]:
SELECT TOP 10 i.industry_name, i.industry_code, j.value
FROM industry AS i
JOIN series AS s
ON i.industry_code = s.industry_code
JOIN january_2017 AS j 
ON s.series_id = j.series_id
WHERE series_title = 'Aggregate weekly payrolls of production and nonsupervisory employees' 
ORDER BY j.value DESC;

SELECT TOP 10 i.industry_name, i.industry_code, j.value
FROM industry AS i
JOIN series AS s
ON i.industry_code = s.industry_code
JOIN january_2017 AS j 
ON s.series_id = j.series_id
WHERE series_title = 'Aggregate weekly payrolls of production and nonsupervisory employees' 
ORDER BY j.value ASC;

(10 rows affected)

(10 rows affected)

Total execution time: 00:00:00.112

industry_name industry_code value Total private 5000000 74498171 Total private 5000000 74498171 Total private 5000000 73474302 Total private 5000000 73474302 Private service-providing 8000000 60914722 Private service-providing 8000000 60914722 Private service-providing 8000000 60807771 Private service-providing 8000000 60807771 Professional and business services 60000000 15251268 Professional and business services 60000000 15251268

industry_name industry_code value Coin-operated laundries and drycleaners 80812310 10079 Coin-operated laundries and drycleaners 80812310 10079 Coin-operated laundries and drycleaners 80812310 10145 Coin-operated laundries and drycleaners 80812310 10145 Skiing facilities 70713920 10564 Skiing facilities 70713920 10564 Marinas 70713930 11560 Marinas 70713930 11560 RV parks and recreational camps 70721200 11598 RV parks and recreational camps 70721200 11598

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [131]:
SELECT s.series_id, i.industry_code, industry_name, j.value
FROM january_2017 AS j 
JOIN series AS s
ON j.series_id = s.series_id
JOIN industry AS i
ON i.industry_code = s.industry_code
WHERE data_type_code = 82 AND j.value >(SELECT AVG(value) 
FROM annual_2016 AS a
JOIN series AS s
ON a.series_id = s.series_id
JOIN industry AS i
ON i.industry_code = s.industry_code
WHERE data_type_code = 82);

SELECT *
FROM datatype
WHERE data_type_code = 82 OR data_type_code = 30;

(294 rows affected)

(2 rows affected)

Total execution time: 00:00:00.107

series_id industry_code industry_name value CES0500000082 5000000 Total private 74498171 CEU0500000082 5000000 Total private 73474302 CES0500000082 5000000 Total private 74498171 CEU0500000082 5000000 Total private 73474302 CES0600000082 6000000 Goods-producing 13426820 CEU0600000082 6000000 Goods-producing 12711945 CES0600000082 6000000 Goods-producing 13426820 CEU0600000082 6000000 Goods-producing 12711945 CES0800000082 8000000 Private service-providing 60914722 CEU0800000082 8000000 Private service-providing 60807771 CES0800000082 8000000 Private service-providing 60914722 CEU0800000082 8000000 Private service-providing 60807771 CES2000000082 20000000 Construction 5335274 CEU2000000082 20000000 Construction 4724156 CES2000000082 20000000 Construction 5335274 CEU2000000082 20000000 Construction 4724156 CES2023600082 20236000 Construction of buildings 1008035 CEU2023600082 20236000 Construction of buildings 940800 CES2023600082 20236000 Construction of buildings 1008035 CEU2023600082 20236000 Construction of buildings 940800 CES2023700082 20237000 Heavy and civil engineering construction 908374 CES2023700082 20237000 Heavy and civil engineering construction 908374 CES2023800082 20238000 Specialty trade contractors 3421902 CEU2023800082 20238000 Specialty trade contractors 3062211 CES2023800082 20238000 Specialty trade contractors 3421902 CEU2023800082 20238000 Specialty trade contractors 3062211 CES2023820082 20238200 Building equipment contractors 1696080 CEU2023820082 20238200 Building equipment contractors 1621266 CES2023820082 20238200 Building equipment contractors 1696080 CEU2023820082 20238200 Building equipment contractors 1621266 CES3000000082 30000000 Manufacturing 7491723 CEU3000000082 30000000 Manufacturing 7411960 CES3000000082 30000000 Manufacturing 7491723 CEU3000000082 30000000 Manufacturing 7411960 CES3100000082 31000000 Durable goods 4859716 CEU3100000082 31000000 Durable goods 4808368 CES3100000082 31000000 Durable goods 4859716 CEU3100000082 31000000 Durable goods 4808368 CES3133200082 31332000 Fabricated metal products 871298 CEU3133200082 31332000 Fabricated metal products 867242 CES3133200082 31332000 Fabricated metal products 871298 CEU3133200082 31332000 Fabricated metal products 867242 CES3133600082 31336000 Transportation equipment 1257764 CEU3133600082 31336000 Transportation equipment 1242685 CES3133600082 31336000 Transportation equipment 1257764 CEU3133600082 31336000 Transportation equipment 1242685 CES3200000082 32000000 Nondurable goods 2633628 CEU3200000082 32000000 Nondurable goods 2605069 CES3200000082 32000000 Nondurable goods 2633628 CEU3200000082 32000000 Nondurable goods 2605069 CES3231100082 32311000 Food manufacturing 851406 CEU3231100082 32311000 Food manufacturing 842201 CES3231100082 32311000 Food manufacturing 851406 CEU3231100082 32311000 Food manufacturing 842201 CES4000000082 40000000 Trade 14850661 CEU4000000082 40000000 Trade 14732108 CES4000000082 40000000 Trade 14850661 CEU4000000082 40000000 Trade 14732108 CES4142000082 41420000 Wholesale trade 4480134 CEU4142000082 41420000 Wholesale trade 4506110 CES4142000082 41420000 Wholesale trade 4480134 CEU4142000082 41420000 Wholesale trade 4506110 CES4142300082 41423000 Durable goods 2261006 CEU4142300082 41423000 Durable goods 2283094 CES4142300082 41423000 Durable goods 2261006 CEU4142300082 41423000 Durable goods 2283094 CES4142400082 41424000 Nondurable goods 1367413 CEU4142400082 41424000 Nondurable goods 1348136 CES4142400082 41424000 Nondurable goods 1367413 CEU4142400082 41424000 Nondurable goods 1348136 CES4142500082 41425000 Electronic markets and agents and brokers 855291 CEU4142500082 41425000 Electronic markets and agents and brokers 873578 CES4142500082 41425000 Electronic markets and agents and brokers 855291 CEU4142500082 41425000 Electronic markets and agents and brokers 873578 CES4142512082 41425120 Wholesale trade agents and brokers 831319 CEU4142512082 41425120 Wholesale trade agents and brokers 851551 CES414

data_type_code data_type_text 30 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES 82 AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period. <b>Average weekly earnings for 2016 for production and nonsupervisoy employees: $878,510.28. Average weekly earnings for January 2017: $1,782,000.96</b></span>

In [47]:
SELECT SUM(value) AS 'Avg earnings', year, period
FROM annual_2016 AS a
JOIN series AS s
ON a.series_id = s.series_id
WHERE data_type_code = 30
GROUP BY year, period
UNION
SELECT SUM(value) AS 'Avg earnings', year, period
FROM january_2017 AS j
JOIN series AS s
ON j.series_id = s.series_id
WHERE data_type_code = 30
GROUP BY year, period;

(2 rows affected)

(2 rows affected)

Total execution time: 00:00:00.131

Avg earnings year period 878510.2799999996 2016 M13 1782000.9599999967 2017 M01

Avg earnings period 878510.2799999996 M13 1782000.9599999967 M01

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

The average weekly earnings for production and nonsupervisory employees for 2016 was $878,510.28 while the average weekly earnings for production and nonsupervisory employees for January 2017 was $1,782,000,96. This shows an increase of $903,490.68 from 2016 to Jan 2017 which makes the time period in which production and nonsupervisory employees fared better, January 2017.

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

<span style="font-size: 11pt; font-family: Arial; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The top performing industries for January 2017 based on the total weekly payroll of production and nonsupervisory employees:</span>

<span style="font-family: Arial; font-size: 11pt; white-space: pre-wrap;">The industry with the highest total weekly payroll in January 2017 is Total Private at 74,498,171; of Total Private we see the industry Private Service-Providing faired at 60,914,722, Professional and Business Services at 15,251,268, Trade at 14,850,661 and Education and Health Services at 14,816,074. All of the aforementioned industries fared better in Jan 2017 than 2016. For comparison, The lowest performing industry based on total weekly payroll was Coin-Operated Laundries and Dry Cleaners at $10,079.</span>

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

I would like to have this information over two solid years (instead of annual\_2016 and january\_2017, have all of 2017). I would like a more indepth, comprehensive and specific breakdown of the industries (as opposed to 'Total Private'), What are the jobs people are working and where?, How many jobs people are working? and What the average salaries for these workers are according to industry.